File: trip_purpose_prediction/v2/get/Python SDK/trip_purpose_prediction.py
# Install the Python library from https://pypi.org/project/amadeus
from amadeus import Client, ResponseError

amadeus = Client(
    client_id='YOUR_AMADEUS_API_KEY',
    client_secret='YOUR_AMADEUS_API_SECRET'
)

try:
    '''
    The passenger is traveling for leisure or business?
    '''
    response = amadeus.travel.predictions.trip_purpose.get(originLocationCode='ATH', destinationLocationCode='MAD',
                                                           departureDate='2020-08-01', returnDate='2020-08-12',
                                                           searchDate='2020-06-11')
    print(response.data)
except ResponseError as error:
    raise error

File: trip_purpose_prediction/v2/get/Node SDK/trip_purpose_prediction.js
const Amadeus = require("amadeus");

const amadeus = new Amadeus({
  clientId: "YOUR_API_KEY",
  clientSecret: "YOUR_API_SECRET",
});

async function main() {
  try {
    // The passenger is traveling for leisure or business?
    const response = await amadeus.travel.predictions.tripPurpose.get({
      originLocationCode: "NYC",
      destinationLocationCode: "MAD",
      departureDate: "2022-08-01",
      returnDate: "2022-08-12",
    });

    console.log(response);
  } catch (error) {
    console.error(error);
  }
}

main();
